app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -277,10 +277,11 @@ private void doMenuAction(int menuItemId) {
                 smoothActionBarToggle.runWhenIdle(new Runnable() {
                     @Override
                     public void run() {
-                        startActivity(intent);
                         overridePendingTransition(R.anim.slide_in_left, R.anim.slide_out_left);
                     }
                 });
+                startActivity(intent);
```

Comment:
Yes. The runnable executes the transition. If the intent is called within the runnable then the activity will come after the transition is up.

You can try it out with the other options on the menu. When you press the button then the page immediately comes up, but as of now if you press Settings, there's a noticeable lag, and it first goes back to where you were and then the SettingsActivity comes . When I started the intent outside the runnable, it worked fine as I wanted. You can check or I can show you a GIF.

No problem though, I can remove the intent and put it back where it was. But the break statement had to be there IMO. @creativepsyco